src/target/imxrt.c

```diff
@@ -50,10 +50,34 @@
 #define IMXRT_SRC_BOOT_MODE1 (IMXRT_SRC_BASE + 0x004U)
 #define IMXRT_SRC_BOOT_MODE2 (IMXRT_SRC_BASE + 0x01cU)
 
-#define IMXRT_OCRAM1_BASE  UINT32_C(0x20280000)
-#define IMXRT_OCRAM1_SIZE  0x00080000U
-#define IMXRT_OCRAM2_BASE  UINT32_C(0x20200000)
-#define IMXRT_OCRAM2_SIZE  0x00080000U
+#define IMXRT_OCRAM1_BASE UINT32_C(0x20280000)
+#define IMXRT_OCRAM1_SIZE 0x00080000U
+#define IMXRT_OCRAM2_BASE UINT32_C(0x20200000)
+#define IMXRT_OCRAM2_SIZE 0x00080000U
+
+#define IMXRT117x_PART_ID 0x88c6U
+
+#define IMXRT117x_ITCM_BASE       0x00000000U /* FlexRAM */
+#define IMXRT117x_ITCM_SIZE       0x00080000U
+#define IMXRT117x_DTCM_BASE       0x20000000U /* FlexRAM */
+#define IMXRT117x_DTCM_SIZE       0x00080000U
+#define IMXRT117x_CODE_TCM_BASE   0x1ffe0000U
+#define IMXRT117x_CODE_TCM_SIZE   0x00020000U
+#define IMXRT117x_SYS_TCM_BASE    0x20000000U
+#define IMXRT117x_SYS_TCM_SIZE    0x00020000U
```

Comment:
Due to how the TCMs work, we had intentionally omitted them from the memory maps before. Are they necessary for breakpoints to work properly, or can they remain omitted on account of faulting till enabled and not being readable in all situations when they are?